File: _examples/cpp20/tuples.cpp
#include <tuple>

// 创建元组
auto person = std::tuple{25, "Alice", 1.68};

// C++20 结构化绑定
auto [age, name, height] = person;

// 使用 std::apply 解包
std::apply([](auto... args) {
    ((std::cout << args << " "), ...);
}, person);

File: _examples/objectivec/tuples.m
// Objective-C 没有原生元组支持
// 通常使用 NSArray 或自定义类
NSArray *person = @[@25, @"Alice", @1.68];
int age = [person[0] intValue];
NSString *name = person[1];

// 或使用字典
NSDictionary *tuple = @{
    @"age": @25,
    @"name": @"Alice",
    @"height": @1.68
};